skrub/_table_vectorizer.py

```diff
@@ -495,147 +435,101 @@ def _clone_transformers(self):
             self.n_jobs,
         )
 
-    def _auto_cast(self, X: pd.DataFrame) -> pd.DataFrame:
-        """Takes a dataframe and tries to convert its columns to their best possible
-        data type.
+    def auto_cast(self, X, reset=True):
```

Comment:
why make it public?

Reply:
While writing it, I thought it might be helpful for the user. Now I realize it will just bring confusion, so I'll revert.